components/malloc_size_of/lib.rs

```diff
@@ -782,3 +791,91 @@ impl MallocSizeOf for xml5ever::QualName {
         self.local.size_of(ops)
     }
 }
+
+impl MallocSizeOf for hyper::header::Headers {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        let mut t = 0;
+        for x in self.iter() {
+           t += x.size_of(ops)
+        }
+        t
+    }
+}
+
+impl<'a> MallocSizeOf for hyper::header::HeaderView<'a> {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        // Close enough?
+        self.name().size_of(ops) + self.value_string().size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::header::ContentType {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        // size_of::<hyper::header::ContentType>
+        self.0.size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Mime {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        self.0.size_of(ops) +
+        self.1.size_of(ops) +
+        self.2.size_of(ops)
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Attr {
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+        self.len()
+    }
+}
+
+impl MallocSizeOf for hyper::mime::Value {
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+        self.len() // Length of string value in bytes (not the char length of a string)!
+    }
+}
+
+/*
+ *  pub struct Duration {
+ *      secs: i64,
+ *      nanos: i32,
+ *  }
+ */
+impl MallocSizeOf for time::Duration {
```

Comment:
Since MallocSizeOf measures heap usage of members, not the size of the value itself, let's use `malloc_size_of_is_0!(time::Duration)` and `malloc_size_of_is_0!(time::Tm)`.